src/util/records.py

```diff
@@ -1544,30 +1544,21 @@ def replace_record_references_batch(cr, id_mapping, model_src, model_dst=None, r
                             ir.res_id,
                         ],
                     )
-                json_path = cr.mogrify(
-                    "$.* ? ({})".format(" || ".join(["@ == %s"] * len(id_mapping))),
-                    list(id_mapping),
-                ).decode()
-
-                query = cr.mogrify(
-                    format_query(
-                        cr,
-                        """
-                        UPDATE {table}
-                           SET {column} = (
-                                   SELECT jsonb_object_agg(key, COALESCE(((jsonb_object(%s::text[]))->>value)::int, value::int))
-                                   FROM jsonb_each_text({column})
-                               )
-                         WHERE {column} IS NOT NULL
-                           AND {column} @? {json_path}
-                        """,
-                        table=ir.table,
-                        column=ir.res_id,
-                        json_path=sql.Literal(json_path),
-                    ),
-                    [list(map(list, id_mapping.items()))],
-                ).decode()
-                explode_execute(cr, query, table=ir.table)
+                query = format_query(
+                    cr,
+                    """
+                     UPDATE {table} t
+                        SET {column} = (
+                                SELECT jsonb_object_agg(key, COALESCE((SELECT new FROM _upgrade_rrr WHERE old = value::int), value::int))
+                                FROM jsonb_each_text(t.{column}))
+                    FROM _upgrade_rrr map
+                    WHERE t.{column} IS NOT NULL
+                        AND (SELECT * FROM JSONB_PATH_QUERY(t.{column}, '$.*'))::int = map.old
```

Comment:
I don't get how this work? Did you mean `map.old IN ...` Don't we get the error of casting jsonb null type to integer?

Reply:
I think this is also updating the same row multiple times. I'd suggest we collect the info to update in a CTE to avoid that.

Reply:
quick test:
```
> select 1 where (SELECT jsonb_path_query FROM JSONB_PATH_QUERY('{"1":3, "2":4, "3":null}'::jsonb, '$.*'))::int = 1
more than one row returned by a subquery used as an expression
```

Reply:
seems the data i tried had a single company per record and no nulls, updated the syntax and tested
I also added values that will be overwritten if updated multiple times but they seem to be correct, checking more into that